test/ExecutionFramework.cpp

```diff
@@ -101,7 +101,7 @@ u256 ExecutionFramework::gasPrice() const
 
 u256 ExecutionFramework::blockHash(u256 const& _number) const
 {
-	return {EVMHost::convertFromEVMC(m_evmHost->get_block_hash(uint64_t(_number & numeric_limits<uint64_t>::max())))};
+	return {EVMHost::convertFromEVMC(m_evmHost->get_block_hash(int64_t(_number & numeric_limits<int64_t>::max())))};
```

Comment:
I think this changes behavior.

Not sure how `u256` comes into the equation (it's `boost::multiprecision::number` so there might be something I'm missing) but for example for a normal 8 bits number you would get:
- `255 & numeric_limits<int8_t>::max()` = `255 & 127` = `127`
- `255 & numeric_limits<uint8_t>::max()` = `255 & 255` = `255`